doc/user_manual/logical_model.tex

```diff
@@ -0,0 +1,177 @@
+      <variables>x, y, z</variables>
+    </ExternalModel>
+
+    <ExternalModel ModuleToLoad="minus" name="minus" subType="">
+      <variables>x, y, z</variables>
+    </ExternalModel>
+
+    <ExternalModel ModuleToLoad="multiply" name="multiply" subType="">
+      <variables>x, y, z</variables>
+    </ExternalModel>
+
+    <LogicalModel name="logical" subType="">
+      <Model class="Models" type="ExternalModel">sum</Model>
+      <Model class="Models" type="ExternalModel">minus</Model>
+      <Model class="Models" type="ExternalModel">multiply</Model>
+      <ControlFunction class="Functions" type="External">control</ControlFunction>
+    </LogicalModel>
+  </Models>
+  ...
+  <Steps>
+    <MultiRun name="mc">
+      <Input class="DataObjects" type="PointSet">inputHolder</Input>
+      <Model class="Models" type="LogicalModel">logical</Model>
+      <Sampler class="Samplers" type="MonteCarlo">MonteCarlo</Sampler>
+      <Output class="DataObjects" type="PointSet">outSet</Output>
+      <Output class="DataObjects" type="PointSet">tagetSet</Output>
+      <Output class="OutStreams" type="Print">dumpOut</Output>
+    </MultiRun>
+  </Steps>
+  ...
+</Simulation>
+
+\end{lstlisting}
+
+Corresponding Python function for \xmlNode{ControlFunction}:
+\begin{lstlisting}[language=python]
+def evaluate(self):
+  """
+    Method required by RAVEN to run this as an external model.
+    @ In, self, object, object to store members on
+    @ Out, model, str, the name of external model that
+      will be executed by hybrid model
+  """
+  model = None
+  if self.x > 0 and self.y >1:
+    model = 'sum'
+  elif self.x > 0 and self.y <= 1:
+    model = 'multiply'
+  else:
+    model = 'minus'
+  return model
+\end{lstlisting}
+
+\textbf{Example (Code):}
+\begin{lstlisting}[style=XML,morekeywords={subType,debug,repeat,name,class,type}]
+<Simulation>
+  ...
+  <Models>
+    <Code name="poly" subType="GenericCode">
+      <executable>logicalCode/poly_code.py</executable>
+      <clargs arg="python" type="prepend"/>
+      <clargs arg="-i" extension=".one" type="input"/>
+      <fileargs arg="aux" extension=".two" type="input"/>
+      <fileargs arg="output" type="output"/>
+    </Code>
+    <Code name="exp" subType="GenericCode">
+      <executable>logicalCode/exp_code.py</executable>
+      <clargs arg="python" type="prepend"/>
+      <clargs arg="-i" extension=".one" type="input"/>
+      <fileargs arg="aux" extension=".two" type="input"/>
+      <fileargs arg="output" type="output"/>
+    </Code>
+    <LogicalModel name="logical" subType="">
+      <Model class="Models" type="Code">poly</Model>
+      <Model class="Models" type="Code">exp</Model>
+      <ControlFunction class="Functions" type="External">control</ControlFunction>
+    </LogicalModel>
+  </Models>
+  ...
+  <Steps>
+    <MultiRun name="logicalModelCode">
+      <Input class="Files" type="">gen.one</Input>
+      <Input class="Files" type="">gen.two</Input>
```

Comment:
Are you fine if I work on this issue in another pull request? Since the changes will also affect EnsembleModel and CodeInterface Entities. It may take longer and further discussion on the solution for this issue.

Reply:
Yes. Please, reference a new issue about this.
Thank you.

Reply:
Issue #1240 is created.